lib/screens/UsersProfileScreen.dart

```diff
@@ -0,0 +1,1576 @@
+                              },
+                              textInputAction: TextInputAction.next,
+                            ),
+                          ),
+                        ],
+                      ),
+                      Column(
+                        children: <Widget>[
+                          Container(
+                            margin: const EdgeInsets.only(top: 15),
+                            child: const Text(
+                              'Enter Password to confirm changes',
+                              style: TextStyle(
+                                fontSize: 18,
+                                color: Colors.black,
+                              ),
+                              textAlign: TextAlign.center,
+                            ),
+                          ),
+                          Container(
+                            width: 320,
+                            margin: const EdgeInsets.only(bottom: 10),
+                            padding: const EdgeInsets.symmetric(
+                                vertical: 5, horizontal: 5),
+                            decoration: const BoxDecoration(
+                              color: mainScheme,
+                              borderRadius:
+                              BorderRadius.all(Radius.circular(10)),
+                            ),
+                            child: TextField(
+                              maxLines: 1,
+                              controller: _password,
+                              obscureText: true,
+                              decoration: unfilledPassword
+                                  ? globalDecoration.copyWith(
+                                  enabledBorder: const OutlineInputBorder(
+                                      borderSide:
+                                      BorderSide(color: Colors.red)),
+                                  suffixIcon: const Icon(Icons.clear,
+                                      color: Colors.red),
+                                  hintText: 'Enter Password')
+                                  : globalDecoration.copyWith(hintText: 'Enter Password'),
+                              style: const TextStyle(
+                                fontSize: 20,
+                                color: Colors.black,
+                              ),
+                              textAlign: TextAlign.left,
+                              textInputAction: TextInputAction.next,
+                              onChanged: (password) {
+                                if (password.isEmpty) {
+                                  setState(() => unfilledPassword = true);
+                                }
+                                else {
+                                  setState(() => unfilledPassword = false);
+                                }
+                              },
+                            ),
+                          ),
+                          Container(
```

Comment:
When(/if) you have time, can you look into reusable components? There is a decent amount of duplication with small changes; I imagine you could parameterize stuff so that you can include a child container here and pass in whatever specific stuff (e.g. `hintText`). That would cut down on duplication a lot here and elsewhere - obviously it's not a priority, but if you have time and interest I think it would make the codebase cleaner and be a good learning experience for you (and me, living vicariously through you :laughing: )

Reply:
I have definitely thought about doing this, but was gonna leave it until most of the important stuff was done first. I can make it a sooner rather than later thing though if desired.